### PR TITLE
Update crash reporter

### PR DESCRIPTION
**Pull request by fractalwrench**
Not ready for review yet, opening so that I can review changes and also keep a log of anything that would break an upgrading user:

- Users now need to link their targets (Test app + custom keyboard) with libc++.tbd and libz.tbd (is there a way around this)?
- Only iOS workspace has been updated so far
- Tests are failing

### CODE REVIEW

**Comment by fractalwrench** [comment]
I've updated all the workspaces and fixed the flaky iOS test (an issue with the project.pbxproj file). A summary of changes:

- Removes git submodules
- Removes KSCrash as a pod dependency, and includes the source files within the Bugsnag framework instead
- Reduce Bugsnag dependency on KSCrash to 3 files (BugsnagCrashSentry, BugsnagSink, BugsnagErrorReportApiClient), by creating a couple of wrapper classes/moving functionality into our code
- Fixes performance in #152. This turned out to be an issue with crash report JSON serialisation calling `write` thousands of times, which took about 200ms on an iPhone 7. Using a 128kb buffer has got that down to 27ms. Changes can be seen in the last few commits.

There may be some additional performance gains in allowing the user to configure whether all the thread traces are recorded, as that now seems to be the bottleneck. This could help out on older hardware. However, I think it would be best to open that in a separate ticket for reviewability purposes.

**Comment by fractalwrench** [comment]
This also requires for user to link against libc++.tbd and libz.tbd for a manual installation, I believe we can set this up somehow for Carthage/Cocoapods?

**Comment by kattrali** [comment]
Yep, you'll need to add it to the spec, [like so](https://github.com/kstenerud/KSCrash/blob/master/KSCrash.podspec#L15)

File: Source/BugsnagCrashReport.h [hunk 1]
```diff
@@ -37,9 +37,9 @@ NSString *_Nonnull BSGFormatSeverity(BSGSeverity severity);
 @interface BugsnagCrashReport : NSObject
 
 /**
- *  Create a new crash report from a JSON crash report generated by KSCrash
+ *  Create a new crash report from a JSON crash report generated by BugsnagCrashSentry
```
Comment (kattrali) [inline Source/BugsnagCrashReport.h]: I don't think renaming this makes sense yet, since the underlying interface hasn't changed.

**Comment by kattrali** [comment]
It passes! 🎉

**Comment by fractalwrench** [comment]
@snmaynard I wasn't too sure what needed changing with the Podspec License/License in general. KSCrash also uses MIT - do we need to update our license, or add them as an author? Is there any action we need to take on this for other installation methods (i.e. manual/carthage)

Current Podspec: https://github.com/bugsnag/bugsnag-cocoa/blob/update-crash-reporter/Bugsnag.podspec.json

**Comment by snmaynard** [comment]
The license just says include a copy of it with the source and we are doing that here so we are covered

**Pull request merged**